core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java

```diff
@@ -149,26 +183,71 @@ public void commit(TableMetadata base, TableMetadata metadata) {
     String codecName =
         metadata.property(
             TableProperties.METADATA_COMPRESSION, TableProperties.METADATA_COMPRESSION_DEFAULT);
+    // TODO:This is not compatible with the scenario where the user modifies the metadata file
```

Comment:
No todos in the code. Also not sure what the issue is here

Reply:
假设客户一开始没有压缩metadata.json. 然后在某一天修改了压缩算法,压缩了metadata.json.

Reply:
Let's say the client didn't compress metadata.json at first, and then one day changed the compression algorithm and compressed metadata.json. Then the uncompressed metadata.json will never be cleaned up.

Reply:
> No todos in the code. Also not sure what the issue is here


It looks like this:
```
v1.json  v2.json    v55.json.gz  v56.json.gz  v57.json.gz   ........
```
Because users changed the compression encoding after v2, the metadata from v1 and v2 will never be filtered correctly. They will remain.

Reply:
https://github.com/apache/iceberg/issues/8162

Reply:
Why would we support that? And I notice the TODO still has not been removed

Reply:
I share your thoughts and I don't intend to support it. It's a pain in the arse to deal with and may even introduce some unknown problems. But we should make this clear to users in the documentation.